frontend/app/components/st-time-slot-form/st-time-slot-form.tsx

```diff
@@ -0,0 +1,239 @@
+        }
+        console.log(hrs);
+
+        let meridies = "am";
+
+        if (timeMode === 12) {
+          if (hrs >= 12) {
+            meridies = "pm";
+          } else {
+            meridies = "am";
+          }
+
+          if (hrs === 0) {
+            hrs = 12;
+          } else if (hrs > 12) {
+            hrs -= 12;
+          }
+        } else if (myTimeSlot.meridies !== "") {
+          if (myTimeSlot.meridies === "pm") {
+            if (hrs < 12) {
+              hrs += 12;
+            }
+          } else {
+            if (hrs === 12) {
+              hrs = 0;
+            }
+          }
+        }
+
+        const hrString = (hrs).toString();
+        myTimeSlot.setHour(hrString);
+        myTimeSlot.setMeridiem(meridies);
+        changeHour(hrString);
+
+        setLoading(false);
+      }
+    }, 1);
+    return () => mounted = false;
+  }, [props.timeMode]);
+
+  return (
+    <View style={[CONTAINER, style]}>
+      {!loading && (
+        <View style={{
+          ...styles.sideByside,
+          width: (props.timeMode === 12) ? 305 : 230
+        }}>
+          <DropDownPicker
+            items={[
+              { label: "Monday", value: "mon" },
+              { label: "Tuesday", value: "tue" },
+              { label: "Wednesday", value: "wed" },
+              { label: "Thursday", value: "thu" },
+              { label: "Friday", value: "fri" },
+              { label: "Saturday", value: "sat" },
+              { label: "Sunday", value: "sun" },
+            ]}
+            defaultValue={props.timeSlot.day}
+            containerStyle={styles.container}
+            style={{ backgroundColor: pickerColor }}
+            itemStyle={
+              styles.flexStart
+            }
+            dropDownStyle={{ backgroundColor: pickerColor }}
+            onChangeItem={item => props.timeSlot.setDay(item.value)}
+            placeholderStyle={{fontSize: 16}} 
+            labelStyle={{fontSize: 16}}   
+            arrowSize={16} 
+            arrowColor={darkAqua}
+          />
+          <TextInput
+            testID={"hourInput" + props.index}
+            style={styles.textInputTime1}
+            onChangeText={text => validateHour(text)}
+            placeholder="hh"
+            defaultValue={props.timeSlot.hour}
+            keyboardType = 'numeric'
+            maxLength = {2}
+            value={hour}
+          />
+          <Text style={styles.colon}>:</Text>
+          <TextInput
+            testID={"minInput" + props.index}
+            style={styles.textInputTime2}
+            onChangeText={text => validateMin(text)}
+            placeholder="mm"
+            defaultValue={props.timeSlot.minute}
+            keyboardType = 'numeric'
+            maxLength = {2}
+            value={min}
+          />
+          {props.timeMode == 12 && (
+            <SwitchSelector
+              style={{width: 70, marginLeft:spacing[2], marginTop:spacing[1]*0.5}}
+              height={34}
+              initial={0}
```

Comment:
```suggestion
              initial={(props.timeSlot.meridies === "pm") ? 1 : 0}
```

This will initialize the slider properly